src/Sulu/Bundle/AudienceTargetingBundle/EventListener/AudienceTargetingCacheListener.php

```diff
@@ -42,24 +42,28 @@ class AudienceTargetingCacheListener implements EventSubscriberInterface
 
     public function preHandle(CacheEvent $cacheEvent)
     {
-        if (self::TARGET_GROUP_URL === $cacheEvent->getRequest()->getRequestUri()) {
+        $request = $cacheEvent->getRequest();
+
+        // requests like "/_sulu_target_group" or "/_fos_user_context_hash" should be ignored
+        if ($request->attributes->get('internalRequest', false)) {
```

Comment:
is this attribute set anywhere else too? or do we need to set it only in the `AudienceTargetingCacheListener`?

Reply:
i guess the `UserContextListener` sets this too? if not, i would call it `isTargetGroupRequest`

Reply:
Yes this is required to be named `internalRequest` as it need to be match with the attribute the `UserContextListener` is setting here: https://github.com/FriendsOfSymfony/FOSHttpCache/blob/39999a6c60d8bdc756436dd0739202d897d29351/src/SymfonyCache/UserContextListener.php#L225